beacon_node/network/src/sync/tests/range.rs

```diff
@@ -1 +1,275 @@
+    }
+
+    fn update_execution_engine_state(&mut self, state: EngineState) {
+        self.log(&format!("execution engine state updated: {state:?}"));
+        self.sync_manager.update_execution_engine_state(state);
+    }
+
+    fn find_blocks_by_range_request(&mut self, target_peer_id: &PeerId) -> (Id, Option<Id>) {
+        let block_req_id = self
+            .pop_received_network_event(|ev| match ev {
+                NetworkMessage::SendRequest {
+                    peer_id,
+                    request: RequestType::BlocksByRange(_),
+                    request_id: AppRequestId::Sync(SyncRequestId::RangeBlockAndBlobs { id }),
+                } if peer_id == target_peer_id => Some(*id),
+                _ => None,
+            })
+            .expect("Should have a blocks by range request");
+
+        let blob_req_id = if self.after_deneb() {
+            Some(
+                self.pop_received_network_event(|ev| match ev {
+                    NetworkMessage::SendRequest {
+                        peer_id,
+                        request: RequestType::BlobsByRange(_),
+                        request_id: AppRequestId::Sync(SyncRequestId::RangeBlockAndBlobs { id }),
+                    } if peer_id == target_peer_id => Some(*id),
+                    _ => None,
+                })
+                .expect("Should have a blobs by range request"),
+            )
+        } else {
+            None
+        };
+
+        (block_req_id, blob_req_id)
+    }
+
+    fn find_and_complete_blocks_by_range_request(&mut self, target_peer_id: PeerId) {
+        let (blocks_req_id, blobs_req_id) = self.find_blocks_by_range_request(&target_peer_id);
+
+        // Complete the request with a single stream termination
+        self.log(&format!(
+            "Completing BlocksByRange request {blocks_req_id} with empty stream"
+        ));
+        self.send_sync_message(SyncMessage::RpcBlock {
+            request_id: SyncRequestId::RangeBlockAndBlobs { id: blocks_req_id },
+            peer_id: target_peer_id,
+            beacon_block: None,
+            seen_timestamp: D,
+        });
+
+        if let Some(blobs_req_id) = blobs_req_id {
+            // Complete the request with a single stream termination
+            self.log(&format!(
+                "Completing BlobsByRange request {blobs_req_id} with empty stream"
+            ));
+            self.send_sync_message(SyncMessage::RpcBlob {
+                request_id: SyncRequestId::RangeBlockAndBlobs { id: blobs_req_id },
+                peer_id: target_peer_id,
+                blob_sidecar: None,
+                seen_timestamp: D,
+            });
+        }
+    }
+
+    fn create_canonical_block(&mut self) -> SignedBeaconBlock<E> {
+        self.harness.advance_slot();
+
+        let block_root =
+            tokio::runtime::Runtime::new()
+                .unwrap()
+                .block_on(self.harness.extend_chain(
```

Comment:
Maybe consider using `async/await` here instead of creating a new `tokio::Runtime` - might keep things simpler and avoid the extra overhead?

Reply:
Switched to async :+1